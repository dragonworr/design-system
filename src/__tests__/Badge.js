import React from 'react'
import renderer from 'react-test-renderer'
import { Badge, theme } from '..'

describe('Badge', () => {
  test('renders', () => {
    const json = renderer.create(<Badge />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('bg green sets background-color and color', () => {
    const json = renderer.create(<Badge bg="green" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('bg lightGreen sets background-color and color', () => {
    const json = renderer.create(<Badge bg="lightGreen" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green0)
    expect(json).toHaveStyleRule('color', theme.colors.black)
  })

  test('bg red sets background-color and color', () => {
    const json = renderer.create(<Badge bg="red" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('bg lightRed sets background-color and color', () => {
    const json = renderer.create(<Badge bg="lightRed" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red0)
    expect(json).toHaveStyleRule('color', theme.colors.black)
  })

  test('bg orange sets background-color and color', () => {
    const json = renderer.create(<Badge bg="orange" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('bg lightOrange sets background-color and color', () => {
    const json = renderer.create(<Badge bg="lightOrange" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange0)
    expect(json).toHaveStyleRule('color', theme.colors.black)
  })

  test('bg blue sets background-color and color', () => {
    const json = renderer.create(<Badge bg="blue" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('bg lightBlue sets background-color and color', () => {
    const json = renderer.create(<Badge bg="lightBlue" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue0)
    expect(json).toHaveStyleRule('color', theme.colors.black)
  })
})
