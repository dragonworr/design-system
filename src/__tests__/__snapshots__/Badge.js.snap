// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge bg blue sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #007aff;
  color: #fff;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg green sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg lightBlue sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #cdf;
  color: #049;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg lightGreen sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #cec;
  color: #060;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg lightOrange sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #feb;
  color: #a50;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg lightRed sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #fcc;
  color: #800;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg orange sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #fa0;
  color: #001833;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge bg red sets background-color and color 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: #c00;
  color: #fff;
}

<div
  className="c0"
  color="text"
/>
`;

exports[`Badge renders 1`] = `
.c0 {
  border-radius: 99999px;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

<div
  className="c0"
  color="text"
/>
`;
